source/beamline/AMBeamline.cpp

```diff
@@ -192,3 +194,27 @@ void AMBeamline::initializeBeamlineSupport(){
 	AMBeamlineSupport::setBeamlineDetectorSetAPI(AMBeamline::bl());
 	AMBeamlineSupport::setBeamlineSynchronizedDwellTimeAPI(AMBeamline::bl());
 }
+
+AMAction3* AMBeamline::createScanInitializationAction(AMGenericStepScanConfiguration *configuration)
```

Comment:
The advantage to defining these actions in the beamline class is that it removes some beamline-dependence away from the scan configuration.